packages/ui/src/utils.ts

```diff
@@ -0,0 +1,32 @@
+import type { OAuthProvider, Web3Provider } from '@clerk/types';
+
+export const PROVIDER_ICON_MAP: Record<OAuthProvider | Web3Provider, string> = {
+  facebook: 'Facebook',
+  google: 'Google',
+  hubspot: 'HubSpot',
+  github: 'GitHub',
+  tiktok: 'TikTok',
+  gitlab: 'GitLab',
+  discord: 'Discord',
+  twitter: 'Twitter',
+  twitch: 'Twitter',
+  linkedin: 'LinkedIn',
+  linkedin_oidc: 'LinkedIn',
+  dropbox: 'Dropbox',
+  atlassian: 'Atlassian',
+  bitbucket: 'Bitbucket',
+  microsoft: 'microsoft',
+  notion: 'Notion',
+  apple: 'Apple',
+  line: 'Line',
+  instagram: 'Instagram',
+  coinbase: 'Coinbase',
+  spotify: 'Spotify',
+  xero: 'Xero',
+  box: 'Box',
+  slack: 'Slack',
+  linear: 'Linear',
+  x: 'Twitter',
+  enstall: 'Enstall',
+  metamask: 'MetaMask',
+};
```

Comment:
@joe-bell curious what you think we should do here? Needed this mapping to connect from provider => our icon component. There still is a type issue in usage that I ignored for the time being. Wasn't sure if this was the right approach, or create a Connection specific Icon that maps all of these?

Reply:
I reckon we could probably merge together with `constants` – what do you reckon?